docroot/modules/custom/foia_webform/src/FoiaSubmissionServiceApi.php

```diff
@@ -0,0 +1,155 @@
+    $this->httpClient = $httpClient;
+    $this->agencyLookUpService = $agencyLookupService;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ConfigFactoryInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $container->get('http_client'),
+      $container->get('foia_webform.agency_lookup_service'),
+      $container->get('logger.channel.foia_webform')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function sendSubmissionToComponent(WebformSubmissionInterface $webformSubmission, NodeInterface $agencyComponent) {
+    $apiUrl = $agencyComponent->get('field_submission_api');
+
+    if ($apiUrl) {
+      $requestData = $this->assembleRequestData($webformSubmission, $agencyComponent);
+    }
+    else {
+      $this->logger
+        ->error('Unable to submit request via the API. Missing API Submission URL for node %component',
+          [
+            '%component' => $agencyComponent->id(),
+            'link' => $agencyComponent->toLink($this->t('Edit Component'), 'edit-form')->toString(),
+          ]
+        );
+    }
+  }
+
+  /**
+   * Gathers the required fields for the API request.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The FOIA form submission form values.
+   * @param \Drupal\node\NodeInterface $agencyComponent
+   *   The Agency Component node object.
+   *
+   * @return array
+   *   Return the assemble request data in an array.
+   */
+  public function assembleRequestData(WebformSubmissionInterface $webformSubmission, NodeInterface $agencyComponent) {
+    $submissionValues = $this->getSubmissionValues($webformSubmission);
+    $agencyInfo = $this->getAgencyInfo($agencyComponent);
+    $requestValues = array_merge($submissionValues, $agencyInfo);
+
+    return $requestValues;
+  }
+
+  /**
+   * Return the FOIA form submission values as an array.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The Webform submission values.
+   *
+   * @return array
+   *   Returns the submission values as an array.
+   */
+  public function getSubmissionValues(WebformSubmissionInterface $webformSubmission) {
+    return $webformSubmission->getData();
+  }
+
+  /**
+   * Parse the Agency information from the Agency Component.
+   *
+   * @param \Drupal\node\NodeInterface $agencyComponent
+   *   The Agency Component node object.
+   */
+  public function getAgencyInfo(NodeInterface $agencyComponent) {
+    return [
+      'agency' => $this->agencyLookUpService->getAgencyFromComponent($agencyComponent),
+      'agency_component_name' => $agencyComponent->label(),
+    ];
+  }
+
+  /**
+   * Get the metadata for the file attachments.
+   *
+   * @param array $files
+   *   An array containing the file IDs for the file attachments.
+   */
+  public function getAttachmentData(array $files) {
+    if (!empty($files)) {
```

Comment:
Rather than passing an array of files here, we may want to consider passing the webform submission values array instead. Then we can iterate through that to find all fids and replace them with the file data you're retrieving below.